mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m

```diff
@@ -406,22 +414,33 @@ - (void)removeAction:(id)sender {
   NSButton *deleteButton = (NSButton *)sender;
   NSDictionary *info = [self.tableContents objectAtIndex:deleteButton.tag];
   NSString *deleteKeyboardMessage = NSLocalizedString(@"message-confirm-delete-keyboard", nil);
+  NSString *keyboardName = nil;
+
+  if ([info objectForKey:@"HeaderTitle"] != nil) {
+    keyboardName = [info objectForKey:@"HeaderTitle"];
+    [self.deleteAlertView setMessageText:[NSString localizedStringWithFormat:deleteKeyboardMessage, keyboardName]];
+  } else {
+    keyboardName = [info objectForKey:kKMKeyboardNameKey];
+    [self.deleteAlertView setMessageText:[NSString localizedStringWithFormat:deleteKeyboardMessage, keyboardName]];
```

Comment:
If this is identical to l.421, can they be pulled out of the if/else?

Reply:
yes, thanks for spotting that!